staging/src/k8s.io/code-generator/cmd/register-gen/generators/packages.go

```diff
@@ -0,0 +1,96 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package generators
+
+import (
+	"github.com/golang/glog"
+
+	clientgentypes "k8s.io/code-generator/cmd/client-gen/types"
+	"k8s.io/gengo/args"
+	"k8s.io/gengo/generator"
+	"k8s.io/gengo/namer"
+	"k8s.io/gengo/types"
+	"strings"
+)
+
+func NameSystems() namer.NameSystems {
+	return namer.NameSystems{}
+}
+
+func DefaultNameSystem() string {
+	return "public"
+}
+
+func Packages(context *generator.Context, arguments *args.GeneratorArgs) generator.Packages {
+	boilerplate, err := arguments.LoadGoBoilerplate()
+	if err != nil {
+		glog.Fatalf("Failed loading boilerplate: %v", err)
+	}
+
+	packages := generator.Packages{}
+	for _, inputDir := range arguments.InputDirs {
+		pkg := context.Universe.Package(inputDir)
+		gv := clientgentypes.GroupVersion{}
+		{
+			pathParts := strings.Split(pkg.Path, "/")
+			if len(pathParts) < 2 {
+				glog.Errorf("the path of the package must contain the group name and the version, path = %s", pkg.Path)
+				continue
+			}
+			gv.Group = clientgentypes.Group(pathParts[len(pathParts)-2])
+			gv.Version = clientgentypes.Version(pathParts[len(pathParts)-1])
+
+			// it there is a comment of the form "// +groupName=somegroup" or "// +groupName=somegroup.foo.bar.io",
+			// extract the fully qualified API group name from it and overwrite the group inferred from the package path
+			if override := types.ExtractCommentTags("+", pkg.DocComments)["groupName"]; override != nil {
+				groupName := override[0]
+				glog.V(5).Infof("overriding the group name with = %s", groupName)
+				gv.Group = clientgentypes.Group(groupName)
+			}
+		}
+
+		typesToRegister := []*types.Type{}
+		for _, t := range pkg.Types {
+			glog.V(5).Infof("considering type = %s", t.Name.String())
+			for _, typeMember := range t.Members {
+				if typeMember.Name == "TypeMeta" && typeMember.Embedded == true {
```

Comment:
would be interesting to know whether this will give a complete list for kube.

Reply:
Have you double checked what the outcome is in kube?

Reply:
I'm not sure what do you mean by `kube`. Perhaps the internal types ? 

I run the generator for types defined in `https://github.com/kubernetes/kubernetes/tree/master/pkg/apis/apps`

here is the output:
```
....
// Adds the list of known types to Scheme.
func addKnownTypes(scheme *runtime.Scheme) error {
        scheme.AddKnownTypes(SchemeGroupVersion,
                &ControllerRevision{},
                &ControllerRevisionList{},
                &StatefulSet{},
                &StatefulSetList{},
        )
        // AddToGroupVersion allows the serialization of client types like ListOptions.
        v1.AddToGroupVersion(scheme, SchemeGroupVersion)
        return nil
}
```

Reply:
I meant against all external types in k8s.io/api.

Reply:
oh, good idea, I will prepare a pull that demonstrates the outcome.

Reply:
alright, I have something, what do you think :) ?

https://github.com/p0lyn0mial/kubernetes/pull/4

Reply:
I generated `register.go` files one more time, this time only for the external types,

https://github.com/p0lyn0mial/kubernetes/pull/5

Reply:
direct link to the diff I found: https://github.com/p0lyn0mial/api/pull/1/files